src/sentry_value.c

```diff
@@ -1254,3 +1254,75 @@ sentry_event_value_add_stacktrace(sentry_value_t event, void **ips, size_t len)
 
     sentry_event_add_thread(event, thread);
 }
+
+void
+sentry_span_set_tag(sentry_value_t span, const char *tag, const char *value)
+{
+    sentry_value_t tags = sentry_value_get_by_key(span, "tags");
+    if (sentry_value_is_null(tags)) {
+        tags = sentry_value_new_object();
+        sentry_value_set_by_key(span, "tags", tags);
+    }
+
+    char *s = sentry__string_clonen(value, 200);
+    if (s) {
+        sentry_value_set_by_key(tags, tag, sentry__value_new_string_owned(s));
+    } else {
+        sentry_value_set_by_key(tags, tag, sentry_value_new_null());
+    }
+}
+
+void
+sentry_span_remove_tag(sentry_value_t span, const char *tag)
+{
+    sentry_value_t tags = sentry_value_get_by_key(span, "tags");
+    if (!sentry_value_is_null(tags)) {
+        sentry_value_remove_by_key(tags, tag);
+    }
+}
+
+void
+sentry_span_set_data(sentry_value_t span, const char *key, sentry_value_t value)
+{
+    sentry_value_t data = sentry_value_get_by_key(span, "data");
+    if (sentry_value_is_null(data)) {
+        data = sentry_value_new_object();
+        sentry_value_set_by_key(span, "data", data);
+    }
+    sentry_value_set_by_key(data, key, value);
+}
+
+void
+sentry_span_remove_data(sentry_value_t span, const char *key)
+{
+    sentry_value_t data = sentry_value_get_by_key(span, "data");
+    if (!sentry_value_is_null(data)) {
+        sentry_value_remove_by_key(data, key);
+    }
+}
+
+void
+sentry_transaction_set_tag(
+    sentry_value_t transaction, const char *tag, const char *value)
+{
+    sentry_span_set_tag(transaction, tag, value);
+}
+
+void
+sentry_transaction_remove_tag(sentry_value_t transaction, const char *tag)
+{
+    sentry_span_remove_tag(transaction, tag);
+}
+
+void
+sentry_transaction_set_data(
+    sentry_value_t transaction, const char *key, sentry_value_t value)
+{
+    sentry_span_set_data(transaction, key, value);
```

Comment:
I just came across this as well as I added these to the Rust SDK (as part of https://github.com/getsentry/sentry-rust/pull/400)

Do transactions have `data`? Or do they rather have `extra` as inherited from the `Event`?

https://develop.sentry.dev/sdk/event-payloads/transaction/ does not mention either as a matter of fact.

I’m confused and don’t know what to believe anymore.

Reply:
From discussions with @relaxolotl: Every transaction is a span and spans have `data`, so transactions must as well. We're also pretty sure that `data` and `extras` are the same thing.